packages/telemetry/src/slog-to-otel.js

```diff
@@ -0,0 +1,640 @@
+        if (spans.has(key)) {
+          spans.end(key, { status, ...attrs }, status !== 'ok' && status);
+        }
+        break;
+      }
+      case 'cosmic-swingset-bootstrap-block-start': {
+        spans.push(`bootstrap-block`);
+        break;
+      }
+      case 'cosmic-swingset-bootstrap-block-finish': {
+        spans.pop(`bootstrap-block`);
+        while (spans.top()) {
+          // Pop off all our context.
+          spans.pop();
+        }
+        break;
+      }
+      case 'cosmic-swingset-begin-block': {
+        while (spans.top()) {
+          // Reset the stack, if we didn't get all the events.
+          spans.pop();
+        }
+        spans.push(`block ${slogAttrs.blockHeight}`);
+        spans.start(`begin-block`, spans.top());
+        spans.end(`begin-block`);
+        break;
+      }
+      case 'cosmic-swingset-deliver-inbound': {
+        spans.start('deliver-inbound', spans.top());
+        spans.end(`deliver-inbound`);
+        break;
+      }
+      case 'cosmic-swingset-bridge-inbound': {
+        spans.start('bridge-inbound', spans.top());
+        spans.end(`bridge-inbound`);
+        break;
+      }
+      case 'cosmic-swingset-end-block-start': {
+        spans.push(`end-block`);
+        break;
+      }
+      case 'cosmic-swingset-end-block-finish': {
+        spans.pop('end-block');
+        break;
+      }
+      case 'cosmic-swingset-save-chain-start': {
+        spans.push(`save-chain`);
+        break;
+      }
+      case 'cosmic-swingset-save-chain-finish': {
+        spans.pop('save-chain');
+        break;
+      }
+      case 'cosmic-swingset-save-external-start': {
+        spans.push(`save-external`);
+        break;
+      }
+      case 'cosmic-swingset-save-external-finish': {
+        spans.pop('save-external');
+        spans.pop(); // 'block ...'
+        break;
+      }
+      case 'solo-process-kernel-start': {
+        spans.push(`solo-process-kernel`);
+        break;
+      }
+      case 'solo-process-kernel-finish': {
+        spans.pop(`solo-process-kernel`);
+        break;
+      }
+      case 'crank-start': {
+        const [name, crankAttrs, links] = extractMessageAttrs(
+          slogAttrs.message,
+        );
+        spans.startNamed(name, getCrankKey(true), spans.top(), crankAttrs, {
+          links,
+        });
+        break;
+      }
+      case 'clist': {
+        spans.get(getCrankKey()).addEvent('clist', cleanAttrs(slogAttrs), now);
+        break;
+      }
+      case 'crank-finish': {
+        let attrs;
+        let error;
+        if (crankNumToCause.has(slogAttrs.crankNum)) {
+          const cause = crankNumToCause.get(slogAttrs.crankNum);
+          ({ attrs, error } = cause);
+          crankNumToCause.delete(slogAttrs.crankNum);
+        }
+        spans.end(getCrankKey(), attrs, error);
+        break;
+      }
+      case 'console': {
+        // We don't care about console messages.  They are out of consensus and
+        // can be really huge.
```

Comment:
true, although they may be important to debugging things, if/when we think about telemetry as a debugging tool more than a performance-measuring tool